docs/docs/tutorials/todos.md

```diff
@@ -0,0 +1,665 @@
+---
+sidebar_position: 4
+description: Build a simple "Todos" application.
+---
+
+# Todos 🗒
+
+:::info
+**Difficulty**: 🟠 Intermediate<br/>
+**Length**: 30 minutes
+
+Before getting started, [read the prerequisites](/docs/overview#prerequisites) to make sure your development environment is ready.
+:::
+
+## Overview
+
+In this tutorial, we're going to build an app that exposes a two endpoints which allow us to perform C.R.U.D operations on a list of todos.
+
+:::note
+C.R.U.D stands for create, read, update, and delete.
+:::
+
+When we're done, we should have an app that supports the following requests:
+
+```bash
+# Create a new todo
+curl --request POST \
+  --url http://localhost:8080/todos \
+  --header 'Content-Type: application/json' \
+  --data '{
+  "title": "Take out trash"
+}'
+
+# Read all todos
+curl --request GET \
+  --url http://localhost:8080/todos
+
+# Read a specific todo by id
+curl --request GET \
+  --url http://localhost:8080/todos/<id>
+
+# Update a specific todo by id
+curl --request PUT \
+  --url http://localhost:8080/todos/<id> \
+  --header 'Content-Type: application/json' \
+  --data '{
+  "title": "Take out trash!",
+  "isCompleted": true
+}'
+
+# Delete a specific todo by id
+curl --request DELETE \
+  --url http://localhost:8080/todos/<id>
+```
+
+## Creating a new app
+
+To create a new Dart Frog app, open your terminal, `cd` into the directory where you'd like to create the app, and run the following command:
+
+```bash
+dart_frog create todos
+```
+
+You should see an output similar to:
+
+```
+✓ Creating todos (0.1s)
+✓ Installing dependencies (1.7s)
+
+Created todos at ./todos.
+
+Get started by typing:
+
+cd ./todos
+dart_frog dev
+```
+
+## Running the development server
+
+You should now have a directory called `todos` -- `cd` into it:
+
+```bash
+cd todos
+```
+
+Then, run the following command:
+
+```bash
+dart_frog dev
+```
+
+This will start the development server on port `8080`:
+
+```
+✓ Running on http://localhost:8080 (1.3s)
+The Dart VM service is listening on http://127.0.0.1:8181/YKEF_nbwOpM=/
+The Dart DevTools debugger and profiler is available at: http://127.0.0.1:8181/YKEF_nbwOpM=/devtools/#/?uri=ws%3A%2F%2F127.0.0.1%3A8181%2FYKEF_nbwOpM%3D%2Fws
+[hotreload] Hot reload is enabled.
+```
+
+Make sure it's working by opening [http://localhost:8080](http://localhost:8080) in your browser or via `cURL`:
+
+```bash
+curl --request GET \
+  --url http://localhost:8080
+```
+
+If everything succeeded, you should see `Welcome to Dart Frog!`.
+
+## Todos Data Source
+
+### Creating `package:todos_data_source`
+
+Now that we have a running application, we need to define an interface for a todos data source which will be responsible for exposing APIs to perform C.R.U.D operations on a list of todos.
```

Comment:
```suggestion
Now that we have a running application, we need to define an interface for a todos data source which will be responsible for exposing APIs to perform C.R.U.D operations on a list of todos.
```
is interface the best word here?

Reply:
Yeah I think so

Reply:
Shouldn't abstraction be a better fit? We are defining an abstraction first and then the implementation of that abstraction.

Reply:
Good call, I like abstraction better as well 👍